src/custom-components/ColorPickerModal.tsx

```diff
@@ -0,0 +1,163 @@
+import { gamepadSliderClasses, ModalRoot, SliderField } from "../deck-components";
+import { useState, VFC } from "react";
+
+interface ColorPickerModalProps {
+  closeModal: () => void;
+  onConfirm?(HSLString: string): any;
+  title?: string;
+  defaultH?: number;
+  defaultS?: number;
+  defaultL?: number;
+  defaultA?: number;
+}
+
+export const ColorPickerModal: VFC<ColorPickerModalProps> = ({
+  closeModal,
+  onConfirm = () => {},
+  title = "Color Picker",
+  defaultH = 0,
+  defaultS = 100,
+  defaultL = 50,
+  defaultA = 1,
+}) => {
+  const [H, setH] = useState<number>(defaultH);
+  const [S, setS] = useState<number>(defaultS);
+  const [L, setL] = useState<number>(defaultL);
+  const [A, setA] = useState<number>(defaultA);
+
+  return (
+      <ModalRoot
+        bAllowFullSize
+        onCancel={closeModal}
+        onOK={() => {
+          onConfirm(`hsla(${H}, ${S}%, ${L}%, ${A})`);
+          closeModal();
+        }}
+      >
+        <style>
+        {`
+        :root {
```

Comment:
Could this be moved to a classname scope instead of :root? I'd like to avoid applying global variables as they will collide if we ever make a non-modal option for the color picker that can render multiple at a time for example.

Reply:
Actually wait nevermind that would still cause conflicts so it doesn't matter as much, though I'd still prefer these be scoped to a class for simplicity.

Reply:
> Actually wait nevermind that would still cause conflicts so it doesn't matter as much, though I'd still prefer these be scoped to a class for simplicity.

@beebls I'll be making this a required change before this can be merged to the lib. Go ahead and rebase and make the commits as needed.